Create individual AR storybook and chromatic

joecowton1 opened the pull request:
<!-- In this repo you can label a PR with the "PR Deployment" label to deploy the code to a publicly accessible url -->
## What does this change?

This is the third and final PR for creating individual storybooks, following on from #6590  and #6561

Details:

- add chromatic config for AR (we'll need to add a required check for `UI tests: APPS_RENDERING`
- adds individual storybook config for AR
- composes AR storybook into composed storybooks
- removes shared storybook config
- adds missing dependencies
- sorts out storybook scripts

georgeblahblah reviewed (commented):
Thanks for this Joe!

I had one comment about what I think is a missing `mobile` breakpoint.

Additionally, I'm not 100% sure that the rendering of components in storybook is now like-for-like with the existing setup on `main`. I feel it's some combination of fonts/styling causing the differences, but I'm very happy to help investigate further.

## Examples

### Newsletter signup

| Storybook on `main` | Storybook on this branch |
|--------|-------|
| ![](https://snapshots.chromatic.com/snapshots/5dfcbf3012392c0020e7140b-63847e067a8cbde0ae5e17fb/capture.png)  |  ![](https://snapshots.chromatic.com/snapshots/637e406971a9af18ddba0505-6382537aca20655bd71cba52/capture.png)  |

### Footer

| Storybook on `main` | Storybook on this branch |
|--------|-------|
| ![](https://snapshots.chromatic.com/snapshots/5dfcbf3012392c0020e7140b-63847e067a8cbde0ae5e17b1/capture.png)  |  ![](https://snapshots.chromatic.com/snapshots/637e406971a9af18ddba0505-6382537aca20655bd71cba08/capture.png)  |

### Rich link

| Storybook on `main` | Storybook on this branch |
|--------|-------|
| ![](https://snapshots.chromatic.com/snapshots/5dfcbf3012392c0020e7140b-63847e067a8cbde0ae5e1806/capture.png)  |  ![](https://snapshots.chromatic.com/snapshots/637e406971a9af18ddba0505-6382537aca20655bd71cba5d/capture.png)  |

georgeblahblah commented:
I think we could also delete `apps-rendering/.storybook/mocks/log4js.js`!

georgeblahblah commented:
Following up on the presentational diff with this storybook, it looks like it's being caused by missing fonts. `preview-head` loads fonts from an `/assets` path which is missing in storybook.

The fonts in DCR's storybook are loaded from `assets.guim` and we should probably look at doing the same for AR's storybook. Ideally AR would also load fonts from `assets.guim`, as captured in https://github.com/guardian/dotcom-rendering/issues/4486.

joecowton1 commented on apps-rendering/.storybook/preview.js:
```diff
@@ -0,0 +1,73 @@
+
+setCookie({ name: 'bwid', value: 'mockBrowserId' });
+
+const guardianViewports = {
+	mobileMedium: {
```
Thats a great idea re source-foundations - maybe one for another PR?

For now i've just added in the suggestion of `mobile`

joecowton1 commented on apps-rendering/.storybook/preview.js:
```diff
@@ -0,0 +1,73 @@
+	},
+};
+
+export const viewports = [375, 480, 660, 740, 980, 1140, 1300];
```
added!

joecowton1 commented:
> Following up on the presentational diff with this storybook, it looks like it's being caused by missing fonts. `preview-head` loads fonts from an `/assets` path which is missing in storybook.
> 
> The fonts in DCR's storybook are loaded from `assets.guim` and we should probably look at doing the same for AR's storybook. Ideally AR would also load fonts from `assets.guim`, as captured in #4486.

I've added a `fonts-css` file similar to our fix for [common-rendering](https://github.com/guardian/dotcom-rendering/pull/6599). Going forward we spoke about using GUIM to host AR fonts permanently. 

I've also added the css reset from source-foundations to fix some styling issues.
